JavaScript/testing.md

```diff
@@ -0,0 +1,109 @@
+# Testing Philosophy
+
+>>> This was originally written for React Native, but the same principles can be applied to the web.
+
+Bugs come in many shapes and sizes. But in actuality, bugs can be broken down into a few categories.
+
+## Table of Contents
+
+- [Testing Philosophy](#Testing-Philosophy)
+  - [Table of Contents](#Table-of-Contents)
+  - [Bug Types](#Bug-Types)
+  - [Testing Methodologies](#Testing-Methodologies)
+    - [Testing for Style Bugs](#Testing-for-Style-Bugs)
+    - [Testing for Business Requirements](#Testing-for-Business-Requirements)
+    - [Testing for Data Flow](#Testing-for-Data-Flow)
+    - [Testing for Accidental Code Changes](#Testing-for-Accidental-Code-Changes)
+  - [Testing in Summary](#Testing-in-Summary)
+
+<a name="bug-types"></a>
+
+## Bug Types
+
+**1. Style bugs**
```

Comment:
In the interest of parallel-listing, would it make sense to rename these headers to something like the following:
```
1. Style/UI Bugs
2. Business Logic Bugs
3. Data Flow Bugs
4. Development Artifact (or some other descriptor noun) Bugs
```

Reply:
makes sense!